Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2215,7 +2234,40 @@ else if (!(req instanceof ServerRequestInitSession) && (!hasSession() || !hasDev
             e.printStackTrace();
         }
     }
-    
+
+    private void processNextQueueItemForced() {
```

Comment:
I'm having trouble understanding why we need this. What specifically about the original implementation is not working?

Reply:
This was created to bypass the waitlocks which were only released on Activity lifecycle callback.